### DISCUSSION TITLE
Group questions on the date field by month

### DISCUSSION THREAD

**Discussion by kingling-abb** [OData/AspNetCoreOData#1204] [Q&A]
Filter working：odata/dirs/deliverrecords?filter=month(deliveryDate) eq 3

GroupBy ：odata/dirs/deliverrecords?$apply=**groupby((month(deliveryDate)), aggregate($count as TotalCount))**

{
    "error": {
        "code": "",
        "message": "The query specified in the URI is not valid. Expression expected at position 28 in 'groupby((month(deliveryDate)), aggregate($count as TotalCount))'.",
        "details": [],
        "innererror": {
            "message": "Expression expected at position 28 in 'groupby((month(deliveryDate)), aggregate($count as TotalCount))'.",
            "type": "Microsoft.OData.ODataException",
            "stacktrace": "   at Microsoft.OData.UriParser.UriQueryExpressionParser.ParseGroupBy()\r\n   at Microsoft.OData.UriParser.UriQueryExpressionParser.ParseApply(String apply)\r\n   at Microsoft.OData.UriParser.ODataQueryOptionParser.ParseApplyImplementation(String apply, ODataUriParserConfiguration configuration, ODataPathInfo odataPathInfo)\r\n   at Microsoft.OData.UriParser.ODataQueryOptionParser.ParseApply()\r\n   at Microsoft.AspNetCore.OData.Query.ApplyQueryOption.get_ApplyClause()\r\n   at Microsoft.AspNetCore.OData.Query.Validator.ODataQueryValidator.Validate(ODataQueryOptions options, ODataValidationSettings validationSettings)\r\n   at Microsoft.AspNetCore.OData.Query.ODataQueryOptions.Validate(ODataValidationSettings validationSettings)\r\n   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.ValidateQuery(HttpRequest request, ODataQueryOptions queryOptions)\r\n   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.OnActionExecuting(ActionExecutingContext actionExecutingContext)"
        }
    }
}

.NET 8.0
Include="Microsoft.AspNetCore.OData" Version="8.2.5"

How should I write this grouping query?

**Comment by rbalagangadharan** [comment]
Try this `$apply=compute(month(deliveryDate) as deliveryDate)/groupby((deliveryDate),aggregate($count as Count))`

**Reply by kingling-abb** [reply]
Great! There is also a small problem, the name setting is different. The statement is completely correct
like this `apply=compute(month(deliveryDate) as deliveryDateMonth)/groupby((deliveryDateMonth),aggregate($count as Count))`
